exchanges/kucoin/kucoin_websocket.go

```diff
@@ -965,10 +965,12 @@ func (ku *Kucoin) processMarketSnapshot(respData []byte, instrument string) erro
 		LastUpdated:  response.Data.Datetime.Time(),
 	}
 	assetEnabledPairs := ku.listOfAssetsCurrencyPairEnabledFor(pair)
-	if assetEnabledPairs[asset.Spot] && ku.AssetWebsocketSupport.IsAssetWebsocketSupported(asset.Spot) && ku.CurrencyPairs.IsAssetEnabled(asset.Spot) == nil {
+	assetEnabledPairsSpot := assetEnabledPairs[asset.Spot] && ku.AssetWebsocketSupport.IsAssetWebsocketSupported(asset.Spot) && ku.CurrencyPairs.IsAssetEnabled(asset.Spot) == nil
+	assetEnabledPairsMargin := assetEnabledPairs[asset.Margin] && ku.AssetWebsocketSupport.IsAssetWebsocketSupported(asset.Margin) && ku.CurrencyPairs.IsAssetEnabled(asset.Margin) == nil
+	switch {
+	case assetEnabledPairsSpot:
 		ku.Websocket.DataHandler <- &spotTickerPrice
-	}
-	if assetEnabledPairs[asset.Margin] && ku.AssetWebsocketSupport.IsAssetWebsocketSupported(asset.Margin) && ku.CurrencyPairs.IsAssetEnabled(asset.Margin) == nil {
+	case assetEnabledPairsMargin:
```

Comment:
Don't make this a case, if its enabled for both spot and margin, we then only populate one with this change

Reply:
changed